<launch>
    
    <node name="left_controller" pkg="pid" type="controller" >
      <param name="Kp" value="0.15" />
      <param name="Ki" value="1.5" />
      <param name="Kd" value="0.0" />
      <param name="upper_limit" value="1" />
      <param name="lower_limit" value="-1" />
      <param name="windup_limit" value="10" />
      <param name="max_loop_frequency" value="20.0" />
      <param name="min_loop_frequency" value="20.0" />
      <remap from="setpoint" to="/left_spd_cmd" />
      <remap from="state" to="/left_spd" />
      <remap from="control_effort" to="/left_vel" />
     </node>
    
    <node name="right_controller" pkg="pid" type="controller" >
      <param name="Kp" value="0.15" />
      <param name="Ki" value="1.5" />
      <param name="Kd" value="0.0" />
      <param name="upper_limit" value="1" />
      <param name="lower_limit" value="-1" />
      <param name="windup_limit" value="10" />
      <param name="max_loop_frequency" value="20.0" />
      <param name="min_loop_frequency" value="20.0" />
      <remap from="setpoint" to="/right_spd_cmd" />
      <remap from="state" to="/right_spd" />
      <remap from="control_effort" to="/right_vel" />
     </node>
    
    <node name="Diff_node" pkg="beobot_control" type="Diff_node">
    </node>


</launch>
